tests/rust/tcp-test/connect/mod.rs

```diff
@@ -86,6 +92,59 @@ fn connect_unbound_socket(libos: &mut LibOS, remote: &SocketAddrV4) -> Result<()
     Ok(())
 }
 
+/// Attempts to connect a TCP socket to a remote that is not accepting connections.
+fn connect_to_bad_remote(libos: &mut LibOS) -> Result<()> {
+    println!("{}", stringify!(connect_to_bad_remote));
+
+    // Create an unbound socket.
+    let sockqd: QDesc = libos.socket(AF_INET, SOCK_STREAM, 0)?;
+
+    // Bad remote address (any localhost port).
+    let remote: SocketAddrV4 = {
+        let ipv4: Ipv4Addr = Ipv4Addr::UNSPECIFIED;
+        SocketAddrV4::new(ipv4, 0)
+    };
+
+    // Succeed to connect socket.
+    let qt: QToken = libos.connect(sockqd, remote)?;
+
+    // Poll for enough time to get the connection refused.
+    match libos.wait(qt, Some(Duration::from_secs(75))) {
```

Comment:
Did you put a timeout on this just to keep the test from hanging? If so, we may want to check explicitly for time out for debugging to see why that happened.

Reply:
This test is about attempting to connect to a TCP socket to a remote that is NOT accepting connections. The only way to be sure that the remote is NOT accepting connections is to wait for 75 seconds and get a `ECONNREFUSED` error. If we get `Err(_)` the test should not pass. Are you suggesting to print the error value?

Reply:
We should probably get a response in less than 75 seconds. If we get a ETIMEOUT after 75 seconds, something else is wrong, so I'm saying that we should check specifically for timeout vs other errors.

Reply:
Are you suggesting something like this?

```rust
    // Poll for enough time to get the connection refused.
    match libos.wait(qt, Some(Duration::from_secs(75))) {
        Ok(qr) if qr.qr_ret == libc::ECONNREFUSED => {},
        Ok(_) => anyhow::bail!("wait() should return ECONNREFUSED"),
        Err(e) if e.errno == libc::ETIMEDOUT => anyhow::bail!("wait() should not timeout"),
        Err(_) => anyhow::bail!("wait() should not fail"),
    }
```

Reply:
Yes!

Reply:
Sure.